File: 2022/14/java/Main.java
import java.io.IOException;

public class Main {
	private static final String INPUT_FILE = "../input/input.txt";

	public static void main(String[] args) throws IOException {
		Map map = Map.load(INPUT_FILE);
		map.simulate();
		System.out.println(map.toString());
		long total = map.countSandTiles();
		System.out.format("1. Total units of sand: %,d%n", total);
	}
}

File: 2022/14/rust/main.rs
pub mod map;

use map::Map;

// --- Constants ---
const INPUT_FILE:&'static str = "../input/input.txt";

fn main() -> Result<(), Box<dyn std::error::Error>> {
    // --- Parse and read the input file ---
    let mut map: Map = Map::load(INPUT_FILE)?;

    // --- Puzzle 1 ---
    map.simulate(1);
    let total = map.count_sand_tiles();
    println!("1. Total units of sand: {}", total);

    // --- Puzzle 2 ---
    map.simulate(2);
    let total = map.count_sand_tiles();
    println!("2. Total units of sand: {}", total);

    Ok(())
}
